src/Nihala.py

```diff
@@ -0,0 +1,398 @@
+    '''
+    m -> masa nihala [kg]
+    k -> koeficient vzmeti 
+    b -> koeficient dušenja
+    t -> čas pri katerem želimo odčitati odmik nihala [s]
+    inital_angle -> začetni odmik nihala [rad]
+    initial_speed -> začetna kotna hitrost nihala [rad/s]
+    n -> število podintervalu na območnu [0,t]
+    '''
+    h = t / (n - 1)  # Velikost koraka
+    time = np.linspace(0, t, n)  # Časovna os
+
+
+    y = np.zeros((n, 2))
+    y[0] = [initial_position, initial_speed]  # Dodajanje začetnih pogojev
+
+    # Runge-Kutta metoda 4. reda
+    for i in range(n-1): 
+        k1 = h * f_damped(time[i], y[i], b,  k, m)
+        k2 = h * f_damped(time[i] + h/2, y[i] + k1/2, b, k, m)
+        k3 = h * f_damped(time[i] + h/2, y[i] + k2/2, b, k, m)
+        k4 = h * f_damped(time[i] + h, y[i] + k3, b, k, m)
+
+        # Update next value of y 
+        y[i+1] = y[i] + (k1 + 2 * k2 + 2 * k3 + k4) / 6.0 
+
+    theta_values = y[:, 0]  # Vrednsoti - odmik [m]
+    omega_values = y[:, 1]  # Vrednsoti - hitrost [m/s]
+
+    
+    w0 = np.sqrt(k/m) # Začetna frekvenca nihanja
+
+    theta_values_energy = np.zeros(len(theta_values))
+    omega_values_energy = np.zeros(len(omega_values))
+    energy = np.zeros(len(omega_values))
+
+    for i in range(0,len(theta_values)):
+        # Računanje vrednosti potencialne energija [J]
+        # odmik^2 * osnovna_frekvenca^2 * masa * 0.5
+        theta_values_energy[i] = theta_values[i] * theta_values[i] * 0.5 * m * w0*w0
+
+    for i in range(0,len(omega_values)):
+        # Računanje vrednosti kinetične energija [J]
+        # hitrost^2 * masa * 0.5
+        omega_values_energy[i] = omega_values[i] * omega_values[i] * 0.5 * m
+
+    for i in range(0,len(energy)):
+        energy[i] = theta_values_energy[i] + omega_values_energy[i]
+
+
+    peaks = np.zeros(len(omega_values))
+    peaks[0] = theta_values[0]
+    parameter = 0
+
+    # Iz izračunanega niza odmikov izluščimo "vrhove" - maksimume
+    for i in range(1,len(theta_values)-1):
+        if(theta_values[i] > 0 ):
+            if(theta_values[i] > theta_values[i+1]):
+                if(parameter == 0):
+                    peaks[i] = theta_values[i]
+            else:
+                peaks[i] = 0
+                parameter = 0
+        else:
+            peaks[i] = 0
+        
+        if(peaks[i-1] > 0):
+            peaks[i] = 0
+            parameter = 1
+
+
+    period = []
+    time_Table = []
+    energy_table = []
+
+    # Glede na ploožaj vrhovov odčitamo čas in velikost energije ko so se pojavili 
+    for i in range(len(peaks)):
+        if(peaks[i] > 0):
+            period.append(peaks[i])
+            time_Table.append(time[i])
+            energy_table.append(energy[i])
+
+
+    # Iz izluščenih maksimimov in časov ko so se pojavili izračunamo nihanji čas vsake periode nihanja
+    oscillation_time = np.zeros(len(period))
+    for i in range(len(period)-1):
```

Comment:
Natančnost izračuna perioda je po vaši metodi omejena s korakom metode. Na ta način ni mogoče doseči visoke natančnosti (ker za povečanje natančnosti potrebujemo vedno manjši korak in posledično vedno več korakov).

Večjo natančnost bi lahko dosegli, če bi zaporedne vrhove iskali kot rešitve enačbe omega=0 in bi uporabili bisekcijo ali Newtonovo metodo.